tests/game/test_buildings.py

```diff
@@ -218,3 +220,393 @@ def test_tree_wildanimalfood_production(s, p):
+	"""
+	settlement, island = settle(s)
+	
+	assert Build(BUILDINGS.FARM, 30, 30, island, settlement=settlement)(p)
+	assert Build(BUILDINGS.PASTURE, 26, 30, island, settlement=settlement)(p)
+	
+	weaver = Build(BUILDINGS.WEAVER, 30, 26, island, settlement=settlement)(p)
+	
+	assert weaver
+	
+	assert weaver.get_component(StorageComponent).inventory[RES.WOOL] == 0
+	assert weaver.get_component(StorageComponent).inventory[RES.TEXTILE] == 0
+	s.run(seconds=60) # 30s pasture, 30s weaver
+	assert weaver.get_component(StorageComponent).inventory[RES.TEXTILE]
+
+
+@game_test()
+def test_pavilion_production(s, p):
+	"""
+	Check whether the pavilion produces faith
+	"""
+	settlement, island = settle(s)
+	pavilion = Build(BUILDINGS.PAVILION, 30, 30, island, settlement=settlement)(p)
+
+	assert pavilion
+	assert pavilion.get_component(StorageComponent).inventory[RES.FAITH] == 0
+	s.run(seconds=30)
+	assert pavilion.get_component(StorageComponent).inventory[RES.FAITH]
+
+
+@game_test()
+def test_farm_production(s, p):
+	"""
+	Check whether fields produce resources and the farm transforms them actual goods
+	"""
+	
+	settlement, island = settle(s)
+	farm = Build(BUILDINGS.FARM, 30, 30, island, settlement=settlement)(p)
+	
+	assert farm
+	
+	primary_resources = (RES.LAMB_WOOL, RES.POTATOES, RES.RAW_SUGAR, RES.TOBACCO_PLANTS, RES.CATTLE, RES.PIGS, RES.HERBS,
+						 RES.GRAIN, RES.SPICE_PLANTS, RES.COCOA_BEANS, RES.VINES, RES.ALVEARIES, RES.HOP_PLANTS)
+
+	secondary_resources = (RES.WOOL, RES.FOOD, RES.SUGAR, RES.TOBACCO_LEAVES, RES.CATTLE_SLAUGHTER, RES.PIGS_SLAUGHTER,
+						   RES.MEDICAL_HERBS, RES.CORN, RES.SPICES, RES.COCOA, RES.GRAPES, RES.HONEYCOMBS, RES.HOPS)
+
+	# TODO: add collectors for handling primary resources transport from fields
+	for a, b in zip(primary_resources, secondary_resources):
+		assert farm.get_component(StorageComponent).inventory[a] == 0
+		farm.get_component(StorageComponent).inventory.alter(a, 2) # 2 potatoes give 1 unit of food
+		
+		s.run(seconds=5)
+		assert farm.get_component(StorageComponent).inventory[b]
+
+
+@game_test()
+def test_school_production(s, p):
+	"""
+	Check whether schools produce education
+	"""
+	settlement, island = settle(s)
+	school = Build(BUILDINGS.VILLAGE_SCHOOL, 30, 30, island, settlement=settlement)(p)
+	
+	assert school
+	
+	assert school.get_component(StorageComponent).inventory[RES.EDUCATION] == 0
+	s.run(seconds=30)
+	assert school.get_component(StorageComponent).inventory[RES.EDUCATION]
+
+
+@game_test()
+def test_saltpond_production(s, p):
+	"""
+	Check whether saltponds produce salt
+	"""
+	settlement, island = settle(s)
+	saltpond = Build(BUILDINGS.SALT_PONDS, 25, 20, island, settlement=settlement)(p)
+	
+	assert saltpond
+	
+	assert saltpond.get_component(StorageComponent).inventory[RES.SALT] == 0
+	s.run(seconds=60)
+	assert saltpond.get_component(StorageComponent).inventory[RES.SALT] >= 2 # ponds produce salt in units of 2
+	
+
+@game_test()
+def test_distillery_production_chain(s, p):
```

Comment:
`tests/game/test_farm.py`  already has a distillery test, both are similar, so I guess just remove the one from that file. See comment above.